src/main.ts

```diff
@@ -157,6 +165,37 @@ function writeRootType(
   options: GenerateOptions,
   w: CodeWriter,
   node: RootType,
+  isLazy: boolean,
+) {
+  return isLazy
+    ? writeLazyRootType(options, w, node)
+    : writeTerminalRootType(options, w, node);
+}
+
+function writeLazyRootType(
+  options: GenerateOptions,
+  w: CodeWriter,
+  node: RootType,
+) {
```

Comment:
Just remembered that this function isn't taking the `export` options into account for the types. Will fix that